Report ruleUID when Script Condition doesn't return a boolean

<!-- Provide a general summary of the feature request in the *Title* above -->
I have a broken rule somewhere with a bogus Script Condition. Unfortunately the error I get is:

    2024-04-09 13:13:59.958 [ERROR] [ernal.handler.ScriptConditionHandler] - Script did not return a boolean value, but 'null'

Looking at the code: https://github.com/openhab/openhab-core/blob/cbb458e0c3c35f353954f82031c5cd1b3943759c/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/ScriptConditionHandler.java#L63

the ruleUID is available. It would be helpful to end users if that ruleUID were included in this and the "Script execution failed" error messages to make identifying the cause of the error easier.

<!-- Important: Please contact the openHAB community forum for questions or -->
<!-- for configuration and usage guidance: https://community.openhab.org -->

## Your Environment
<!-- Include as many relevant details about the environment when applicable -->
* Version used: (e.g., openHAB and add-on versions) OH 4.2 M2
* Environment name and version (e.g. Chrome 111, Java 17, Node.js 18.15, ...): Official Docker image
* Operating System and version (desktop or mobile, Windows 11, Raspbian Bullseye, ...):

--- a/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/ScriptConditionHandler.java
+++ b/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/ScriptConditionHandler.java
@@ -60,10 +60,12 @@ public class ScriptConditionHandler extends AbstractScriptModuleHandler<Conditio
                 if (returnVal instanceof Boolean boolean1) {
                     result = boolean1;
                 } else {
-                    logger.error("Script did not return a boolean value, but '{}'", returnVal);
+                    logger.error("Script of rule with UID '{}' did not return a boolean value, but '{}'", ruleUID,
+                            returnVal);
                 }
             } catch (ScriptException e) {
-                logger.error("Script execution failed: {}", e.getMessage());
+                logger.error("Script execution of rule with UID '{}' failed: {}", ruleUID, e.getMessage(),
+                        logger.isDebugEnabled() ? e : null);
             }
         }
 

